### ISSUE TITLE
overfitting

### ISSUE THREAD

**Issue by JasLin** [cloggy45/Gekko-Bot-Resources#13]
@cloggy45  i just read the source of neuralnet version strategy.

```js
let layers = [
      {type:'input', out_sx:1, out_sy:1, out_depth: 1},
      {type:'fc', num_neurons: this.layer_neurons, activation: this.layer_activation},
      {type:'regression', num_neurons: 1}
    ];
```

in theory, if we only use fully connected layer without doing any thing like ```regularization``` or ```dropout``` , the trained network is easily overfitted

**Comment by JasLin** [comment]
convnetjs haven't implement dropout yet. ref: https://github.com/karpathy/convnetjs/pull/3

**Comment by GO1984** [comment]
are you shure? I found this snippet:
`// dropout half the units (probability 0.5) in this layer during training, for regularization
{type:'fc', num_neurons:10, activation:'relu', drop_prob: 0.5}`
[https://cs.stanford.edu/people/karpathy/convnetjs/docs.html](url)

**Comment by JasLin** [comment]
@GO1984, hi bro, great,  u r right, the ref I had gave is about ```DropConnect```, but not ```dropout```, they are difference things.

maybe we should  add this into the layer design, and backtest it.

**Comment by cloggy45** [comment]
Any changes you would like to make to strategy submit a PR. In the meantime I will close this issue.
